pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -579,36 +583,41 @@ func (o *MustGatherOptions) Run() error {
 	}
 
 	// print at both the beginning and at the end.  This information is important enough to be in both spots.
-	o.PrintBasicClusterState(context.TODO())
+	o.PrintBasicClusterState(ctx)
 	defer func() {
+		// Shortcircuit this in case the context is already cancelled.
+		if ctx.Err() != nil {
+			return
+		}
 		fmt.Fprintf(o.RawOut, "\n\n")
 		fmt.Fprintf(o.RawOut, "Reprinting Cluster State:\n")
-		o.PrintBasicClusterState(context.TODO())
+		o.PrintBasicClusterState(ctx)
 	}()
 
 	// Ensure resource cleanup unless instructed otherwise ...
-	var cleanupNamespace func()
+	var cleanupNamespace func(context.Context)
 	if !o.Keep {
 		defer func() {
 			if cleanupNamespace != nil {
-				cleanupNamespace()
+				cleanupNamespace(context.TODO())
 			}
 		}()
 	}
 
-	// Due to 'stack unwiding', this should happen after 'clusterState' printing, to ensure that we always
-	//  print our ClusterState information.
+	// Due to 'stack unwinding', this should happen after 'clusterState' printing, to ensure that we always
+	// print our ClusterState information.
 	runBackCollection := true
 	defer func() {
-		if !runBackCollection {
+		// Shortcircuit this in case the context is already cancelled.
+		if ctx.Err() != nil || !runBackCollection {
```

Comment:
I assume that once user interrupts the process with SIGTERM, it makes sense to not even collect BackupGathering?

Reply:
I think that the message of a signal is that you want to terminate. So I don't see why to do more work.